apps/modernization-ui/src/apps/page-builder/components/QuestionLibrary/QuestionLibraryTable.tsx

```diff
@@ -0,0 +1,196 @@
+    const [search, setSearch] = useState<string>('');
+    const [selectedQuestion, setSelectedQuestion] = useState<Question>({});
+
+    const { state } = useContext(UserContext);
+    const authorization = `Bearer ${state.getToken()}`;
+
+    // @ts-ignore
+    const asTableRow = (page: Question): TableBody => ({
+        id: page.id,
+        checkbox: true,
+        tableDetails: [
+            {
+                id: 1,
+                title: <div className="page-name">{page?.questionType || page.codeSet}</div> || null
+            },
+            { id: 2, title: <div className="event-text">{page?.uniqueId}</div> || null },
+            {
+                id: 3,
+                title: <div>{page?.uniqueName}</div> || null
+            },
+            {
+                id: 4,
+                title: <div>{page?.subgroup}</div> || null
+            },
+            {
+                id: 5,
+                title:
+                    (
+                        <div className="ds-u-text-align--right">
+                            <Icon.ExpandMore style={{ cursor: 'pointer' }} size={4} color="black" />
+                        </div>
+                    ) || null
+            }
+        ]
+    });
+    const handleSelected = ({ target }: any, item: any) => {
+        if (target.checked) {
+            const value: any = summaries.filter((val: any) => item.id === val.id);
+            setSelectedQuestion(value);
+        } else {
+            setSelectedQuestion({});
+        }
+    };
+    const asTableRows = (pages: PageSummary[] | undefined): TableBody[] => pages?.map(asTableRow) || [];
+
+    /*
+     * Converts header and Direction to API compatible sort string such as "name,asc"
+     */
+    const toSortString = (name: string, direction: Direction): string | undefined => {
+        if (name && direction && direction !== Direction.None) {
+            switch (name) {
+                case Column.Type:
+                    return `questionType,${direction}`;
+                case Column.UniqueId:
+                    return `uniqueId,${direction}`;
+                case Column.UniqueName:
+                    return `uniqueName,${direction}`;
+                case Column.SubGroup:
+                    return `subgroup,${direction}`;
+                default:
+                    return undefined;
+            }
+        }
+        return undefined;
+    };
+
+    useEffect(() => {
+        setTableRows(asTableRows(summaries));
+    }, [summaries]);
+
+    const handleSearch = (search: string, filter: any) => {
+        setSearch(search);
+        toSearch?.(search, filter);
+    };
+
+    useEffect(() => {
+        return () => localStorage.setItem('selectedQuestion', '0');
+    }, []);
+
+    const handleSort = (name: string, direction: Direction): void => {
+        sortChange(toSortString(name, direction));
+    };
+    const handleAddQsntoPage = async () => {
+        // @ts-ignore
```

Comment:
I think its better to set a condition if (id) then do the rest

Reply:
Fixed